### ISSUE TITLE
hasPermission() function always return false

### ISSUE THREAD

**Issue by SamuelY** [psecio/gatekeeper#62]
I find that hasPermission() have wrong in UserModel.php.

the function  return ($perm->id !== null && $**perm->id** === $permId) ? true : false;
should be modified as return ($perm->id !== null && $**perm->permissionId** === $permId) ? true : false;

I have a suggestion

hasPermission() return true only if the requested permission isn't expired

please check it

**Comment by gdespaux** [comment]
I also had this issue. I'm not positive, but I believe the `===` may be the issue. Perhaps one of the id's is returned as an integer, and the other as a string? Changing to `==` allows it to function as expected.

**Comment by marksparrish** [comment]
I agree with @SamuelY.  I think the issue stems from $perm being pulled from the user_permission table.  $perm->id is the id from that table and not the id from the permissions table.  However if you change it as suggested you get the correct id.

**Comment by Derpalus** [comment]
@gdespaux is right, the strict comparison has to be modified to a non-strict comparison. The data from the database is a string and the user might give an integer as input. This can be worked around by giving the input like a string even if it's a number: `$user->hasPermission('23');`
And as @SamuelY says; the fact that hasPermission() doesn't take into consideration that a permission also could be expired is something I think should be implemented as well, it could even be considered a bug.
